File: 1706. Where Will the Ball Fall/Solution.py
from typing import List


class Solution:
    def findBall(self, grid: List[List[int]]) -> List[int]:
        n = len(grid[0])
        res = [0] * n
        for i in range(n):
            j = i
            for row in grid:
                k, j = j, j + row[j]
                if j < 0 or j >= n or row[k] != row[j]:
                    j = -1
                    break
            res[i] = j
        return res

File: 1706. Where Will the Ball Fall/Solution.java
class Solution {
    public int[] findBall(int[][] grid) {
        int n = grid[0].length;
        int[] res = new int[n];
        for (int i = 0; i < n; i++) {
            int j = i;
            for (int[] row : grid) {
                int k = j;
                j += row[j];
                if (j < 0 || j >= n || row[j] != row[k]) {
                    j = -1;
                    break;
                }
            }
            res[i] = j;
        }
        return res;
    }
}
